Calculator/Services/NumberService.cs

```diff
@@ -0,0 +1,54 @@
+﻿namespace Calculator.Services
+{
+    public class NumberService : INumberService
+    {
+        private string[] DefaultDelimiters = { ",", "\n" };
+        private string[] DelimiterSurroundingStrings = { "//", "[", "][", "]" };
+
+        public List<int> ParseNumbers(string numbers)
+        {
+            IEnumerable<string> stringOfNumbers;
+
+            if (numbers.StartsWith("//"))
+            {
+                var delimiters = GetDelimiters(numbers);
+                var numbersSeperatedByDelimiters = GetNumbersSeperatedByDelimiters(numbers);
+                stringOfNumbers = numbersSeperatedByDelimiters.Split(delimiters.ToArray(), StringSplitOptions.RemoveEmptyEntries);
+            }
+            else
+            {
+                stringOfNumbers = numbers.Split(DefaultDelimiters, StringSplitOptions.RemoveEmptyEntries);
+            }
+
+            CheckForNegativeNumbers(stringOfNumbers);
+
+            return stringOfNumbers.Select(x => int.Parse(x)).Where(num => num < 1001).ToList(); 
+        }
+
+        private IEnumerable<string> GetDelimiters(string stringOfNumbersIncludingDelimiters)
+        {
+            var delimitersAndNumbers = stringOfNumbersIncludingDelimiters.Split(DefaultDelimiters[1], StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Not a huge fan of using indexes from your array to avoid having magic strings - you're ending up with a similar problem, you don't actually know what the value is (unless you check the array). How else could you handle this?